src/deploy/hosting/convertConfig.ts

```diff
@@ -96,9 +104,16 @@ export async function convertConfig(
   // rewrites to see if it's necessary.
   const hasBackends = !!deploy.config.rewrites?.some((r) => "function" in r || "run" in r);
 
-  // We need to be able to do a rewrite to an existing function that is may not
-  // even be part of Firebase's control or a function that we're currently
-  // deploying.
+  // We need to be able to do a rewrite to an existing function that may not be
+  // under Firebase's control or a function that we're currently deploying.
+
+  let wantBackends: backend.Backend[] = [];
```

Comment:
You don't need to remember each backend by codebase. Consider merging them all together with

```
const wantBackend = backend.merge(
  ...Object.values(functionsPayload.functions || {}).map(c => c.wantBackend)
)
```

Then you can get rid of for loops later.

Reply:
Done.

Reply:
Done.